stock_helper_delivery/models/stock_picking.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2021 Camptocamp SA (http://www.camptocamp.com)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from odoo import models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def get_picking_with_carrier_from_chain(self):
+        """Returns a picking with a carrier among next chained pickings.
+
+        For non-delivery transfers this method will often return the related
+        delivery.
+        """
+        self.ensure_one()
+        if self.carrier_id:
+            return self
+
+        def get_first_move_dest(moves):
+            for move in moves.move_dest_ids:
+                if move.state not in ("cancel", "done"):
+                    return move
+
+        move_dest = get_first_move_dest(self.move_lines)
+        while move_dest:
+            if move_dest.picking_id.carrier_id:
+                return move_dest.picking_id
```

Comment:
```suggestion
        if move_dest:
            if move_dest.picking_id.carrier_id:
                return move_dest.picking_id
```
At most you'll have 1 record. No need to use "while" here, or am I missing something?

Reply:
it means it jumps on the next move till you get a dest move. It stops either if you find the carrier or if you don't have any dest move

Reply:
Well, if I read get_first_move_dest, it'll return a single record.
What I understand from your explaination is something like this
```python
        def get_first_move_dest(moves):
            for move in moves.move_dest_ids:
                if move.state not in ("cancel", "done"):
                    yield move
                    
        while get_first_move_dest(self.move_lines):
            if move_dest.picking_id.carrier_id:
                return move_dest.picking_id
```
Anyway, I'm probably missing something.
Forget about this